// @mysten/wallet-adapter-react-ui/dist/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  ConnectWalletModal: () => ConnectWalletModal,
  ManageWalletModal: () => ManageWalletModal,
  WalletWrapper: () => WalletWrapper
});
module.exports = __toCommonJS(src_exports);

// src/ConnectWalletModal.tsx
var import_material = require("@mui/material");
var import_react = require("react");
var import_wallet_adapter_react = require("@mysten/wallet-adapter-react");
var import_jsx_runtime = require("react/jsx-runtime");
function ConnectWalletModal(props) {
  let { connected } = (0, import_wallet_adapter_react.useWallet)();
  const [open, setOpen] = (0, import_react.useState)(false);
  const handleClickOpen = () => {
    setOpen(true);
  };
  const handleClose = () => {
    setOpen(false);
  };
  const { wallets, wallet, select, connecting } = (0, import_wallet_adapter_react.useWallet)();
  const handleConnect = (walletName) => {
    select(walletName);
    handleClose();
  };
  const style = {
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    width: 300,
    bgcolor: "background.paper",
    border: "2px solid #000",
    boxShadow: 24,
    p: 4,
    borderRadius: 2
  };
  const connectButtonStyle = {
    borderRadius: 7,
    backgroundColor: "#6fbcf0",
    fontWeight: 600
  };
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_jsx_runtime.Fragment, {
    children: !connected && /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(import_jsx_runtime.Fragment, {
      children: [
        /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_material.Button, {
          style: connectButtonStyle,
          variant: "contained",
          onClick: handleClickOpen,
          children: "Connect To Wallet"
        }),
        /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_material.Modal, {
          open,
          onClose: handleClose,
          children: /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(import_jsx_runtime.Fragment, {
            children: [
              !connecting && /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(import_material.Box, {
                sx: style,
                children: [
                  /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_material.Typography, {
                    id: "modal-modal-title",
                    variant: "h6",
                    component: "h2",
                    align: "center",
                    children: "Select Wallet"
                  }),
                  /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_material.List, {
                    children: wallets.map((wallet2, i) => /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_material.ListItemButton, {
                      onClick: () => handleConnect(wallet2.name),
                      children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_material.ListItemText, {
                        primary: wallet2.name
                      })
                    }, i))
                  })
                ]
              }),
              connecting && /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(import_material.Box, {
                sx: style,
                children: [
                  /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(import_material.Typography, {
                    id: "modal-modal-title",
                    variant: "h6",
                    component: "h2",
                    children: [
                      "Connecting to ",
                      wallet ? wallet.name : "Wallet"
                    ]
                  }),
                  /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_material.CircularProgress, {})
                ]
              })
            ]
          })
        })
      ]
    })
  });
}

// src/ManageWalletModal.tsx
var import_material2 = require("@mui/material");
var import_react2 = require("react");
var import_wallet_adapter_react2 = require("@mysten/wallet-adapter-react");
var import_Settings = __toESM(require("@mui/icons-material/Settings"));
var import_jsx_runtime = require("react/jsx-runtime");
function ManageWalletModal(props) {
  let { connected, disconnect, wallet, getAccounts } = (0, import_wallet_adapter_react2.useWallet)();
  const [open, setOpen] = (0, import_react2.useState)(false);
  const [account, setAccount] = (0, import_react2.useState)("");
  const PK_DISPLAY_LENGTH = 10;
  (0, import_react2.useEffect)(() => {
    if (!connected)
      return;
    getAccounts().then((accounts) => {
      if (accounts && accounts.length) {
        setAccount(accounts[0]);
      }
    });
  }, [wallet, connected, getAccounts]);
  const handleClickOpen = () => {
    setOpen(true);
  };
  const handleClickDisconnect = () => {
    disconnect();
    setOpen(false);
  };
  const handleClose = (value) => {
    setOpen(false);
  };
  const style = {
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    width: 300,
    bgcolor: "background.paper",
    border: "2px solid #000",
    boxShadow: 24,
    p: 4,
    borderRadius: 2
  };
  const ManageWalletButtonStyle = {
    borderRadius: 7,
    backgroundColor: "#6fbcf0",
    fontWeight: 600
  };
  const handleCopyAddress = () => {
    navigator.clipboard.writeText(account);
  };
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_jsx_runtime.Fragment, {
    children: connected && wallet && /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(import_jsx_runtime.Fragment, {
      children: [
        /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(import_material2.Button, {
          color: "primary",
          variant: "contained",
          style: ManageWalletButtonStyle,
          onClick: handleClickOpen,
          children: [
            /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_Settings.default, {}),
            " ",
            account.slice(0, PK_DISPLAY_LENGTH),
            "..."
          ]
        }),
        /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_material2.Modal, {
          open,
          onClose: handleClose,
          children: /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(import_material2.Box, {
            sx: style,
            children: [
              /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_material2.Typography, {
                id: "modal-modal-title",
                variant: "h6",
                component: "h2"
              }),
              /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(import_material2.List, {
                children: [
                  /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_material2.ListItemButton, {
                    onClick: handleClickDisconnect,
                    children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_material2.ListItemText, {
                      primary: "Disconnect"
                    })
                  }),
                  /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_material2.ListItemButton, {
                    onClick: handleCopyAddress,
                    children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_material2.ListItemText, {
                      primary: "Copy Address"
                    })
                  })
                ]
              })
            ]
          })
        })
      ]
    })
  });
}

// src/WalletWrapper.tsx
var import_material3 = require("@mui/material");
var import_jsx_runtime = require("react/jsx-runtime");
var theme = (0, import_material3.createTheme)({
  typography: {
    "fontFamily": `"IBM Plex Sans", sans-serif`
  }
});
function WalletWrapper({}) {
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_jsx_runtime.Fragment, {
    children: /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(import_material3.ThemeProvider, {
      theme,
      children: [
        /* @__PURE__ */ (0, import_jsx_runtime.jsx)(ConnectWalletModal, {}),
        /* @__PURE__ */ (0, import_jsx_runtime.jsx)(ManageWalletModal, {})
      ]
    })
  });
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ConnectWalletModal,
  ManageWalletModal,
  WalletWrapper
});
